File: Swift/new_approach/new_approach.swift
import Foundation

class Reverse {
	func reverse_number(_ num : String) -> String {
		let length : Int = num.count
		let bound : Int = length / 2
		var symmetry = length - 1
		var temp : Character = "a";

		var arr : [Character] = [Character](num)

		for i in 0...bound {
			symmetry = length - 1 - i
			temp = arr[i]
			arr[i] = arr[symmetry]
			arr[symmetry] = temp
		}

		let return_str = String(arr)

		return return_str
	}
}

File: Dart/new_approach/new_approach.dart
import 'dart:io';

class Reverse {
	String reverseNumber(String number) {
		var arr = number.runes.toList();

		var len = arr.length;
		var bound = len ~/ 2;
		var i = 0;
		var symmetry = len - 1 - i;
		var temp;

		for (i = 0; i < bound; i++) {
			symmetry = len - 1 - i;
			temp = arr[i];
			arr[i] = arr[symmetry];
			arr[symmetry] = temp;
		}

		var returnStr = new String.fromCharCodes(arr);
		return returnStr;
	}
}

void main() {
	Reverse rev = new Reverse();
	var number = stdin.readLineSync();
	var res = rev.reverseNumber(number);
	print(res);
}